pkg/indexer/rpc_streamer/rpc_log_streamer.go

```diff
@@ -244,12 +258,44 @@ backfillLoop:
 			logger.Error("Context cancelled, stopping watcher")
 			return
 
+		case err, open := <-sub.Err():
+			if !open {
+				logger.Error("subscription channel closed, closing watcher")
+				return
```

Comment:
does this mean we will stop indexing until the entire process gets restarted?

Reply:
I've removed checking the sub error channel is open. Instead we'll rebuild the sub when there's an actual error. And the sub error channel is recreated by go-ethereum (the owner)